src/main/java/com/devcycle/sdk/server/common/api/DVCApiClient.java

```diff
@@ -1,40 +1,76 @@
 package com.devcycle.sdk.server.common.api;
 
 import com.devcycle.sdk.server.common.interceptor.AuthorizationHeaderInterceptor;
+import com.devcycle.sdk.server.common.model.DVCOptions;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
 import okhttp3.*;
 import retrofit2.Retrofit;
 import retrofit2.converter.jackson.JacksonConverterFactory;
 
 import java.util.Objects;
+import java.util.concurrent.TimeUnit;
 
 public final class DVCApiClient {
 
   private final OkHttpClient.Builder okBuilder;
   private final Retrofit.Builder adapterBuilder;
 
-  private static final String BASE_URL = "https://bucketing-api.devcycle.com/";
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
-  private DVCApiClient() {
-    okBuilder = new OkHttpClient.Builder();
+  private static final String BUCKETING_URL = "https://bucketing-api.devcycle.com/";
+  private static final String CONFIG_URL = "https://config-cdn.devcycle.com/";
+  private static final int DEFAULT_TIMEOUT_MS = 10000;
+  private static final int MIN_INTERVALS_MS = 1000;
+  
+  private Boolean cloudBucketing = false;
+  private String configUrl;
+  private int requestTimeoutMs;
+
+  private DVCApiClient(DVCOptions options) {
+    String url;
 
     String bucketingApiUrl = System.getenv("BUCKETING_API_URL");
-    String url = Objects.isNull(bucketingApiUrl) || Objects.equals(bucketingApiUrl, "") ? BASE_URL : bucketingApiUrl;
+    OBJECT_MAPPER.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    okBuilder = new OkHttpClient.Builder();
 
-    adapterBuilder = new Retrofit
-            .Builder()
-            .baseUrl(url)
-            .addConverterFactory(JacksonConverterFactory.create());
+    String cdnUrlFromOptions = options.getConfigCdnBaseUrl();
+    Boolean enableCloudBucketing = options.getEnableCloudBucketing();
+    int configRequestTimeoutMs = options.getConfigRequestTimeoutMs();
+
+    this.cloudBucketing = Objects.isNull(enableCloudBucketing) ? false : enableCloudBucketing;
```

Comment:
honestly you should make this value default to false instead of doing an object null check on it lol

Reply:
this will default to false? or do you mean to default it to false in the options constructor itself?

Reply:
In java land its really really weird to see that you'd have to do a null check for a boolean like this. 

like just initialize the default as false if you're going to make it be false if its null anyway.